tests/components/fastdotcom/test_diagnostics.py

```diff
@@ -0,0 +1,43 @@
+"""Test the Fast.com component diagnostics."""
+
+from unittest.mock import patch
+
+from syrupy import SnapshotAssertion
+
+from homeassistant.components.fastdotcom.const import DEFAULT_NAME, DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+from tests.typing import ClientSessionGenerator
+
+
+async def test_get_config_entry_diagnostics(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test if get_config_entry_diagnostics returns the correct data."""
+    config_entry = MockConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title=DEFAULT_NAME,
+        source=SOURCE_USER,
+        options={},
+        entry_id="TEST_ENTRY_ID",
+        unique_id="UNIQUE_TEST_ID",
+        minor_version=1,
+    )
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.fastdotcom.coordinator.fast_com", return_value=50.3
+    ):
```

Comment:
We should not patch the coordinator data; instead, we need to patch the underlying library to return mocked values.

Reply:
fast_com is the method patched and used within the [coordinator](https://github.com/erwindouna/ha-core/blob/9eeec391303d5367186e9350569324fc11da3f17/homeassistant/components/fastdotcom/coordinator.py#L31). That's also how the coordinator itself is [tested](https://github.com/home-assistant/core/blob/30d1f70468235bd7ff8d163d43dddecde0b49f87/tests/components/fastdotcom/test_coordinator.py#L28) - which is released. Maybe I am overlooking the point you're making, feel free to elaborate. :)

Reply:
The one overlooking here... is me 🤦 

Sorry about that 👍